backend/models/userModel.js

```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose')
+const bcrypt = require('bcrypt')
+
+const Schema = mongoose.Schema
+
+const userSchema = new Schema({
+    email: {
+        type: String,
+        require: true,
+        unique: true
+    },
+    password: {
+        type: String, 
+        require: true
+    }
+})
+
+// static signup method
+userSchema.statics.signup = async function( email, password ) {
+
+    const exists = await this.findOne({ email })
+
+    if (exists) {
+        throw Error('Email already in use')
+    }
+
+    const salt = await bcrypt.genSalt(10)
```

Comment:
Could you please add a comment explaining what is salt f.e. ("A random string of characters added to the hash as another layer of protection.")

Reply:
As requested three explaining comments added to the encrypting function